File: core/java/src/main/java/hara/lang/protocol/IPushFirst.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.ipushfirst", name = "IPushFirst")
public interface IPushFirst<E> {
  @HaraMethod(value = "push-first", arity = 2)
  IPushFirst<E> pushFirst(E e);
}

File: core/rust/src/lang/protocol/ipushfirst.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.ipushfirst", name = "IPushFirst")]
pub trait IPushFirst<E> {
    type Output;

    #[hara_method(value = "push-first", arity = 2)]
    fn push_first(&self, value: E) -> Self::Output;
}
